### ISSUE TITLE
libwasmedge exposes hundreds of C++ symbols (0.14.0 regression)

### ISSUE THREAD

**Issue by paravoid** [WasmEdge/WasmEdge#3655]
### Summary

Compiling 0.14.0 results into a ton of `WasmEdge::` C++ symbols being exposed. 684 to be exact. 

For example:
``` WasmEdge::PO::ArgumentParser::SubCommandDescriptor::consume_argument(WasmEdge::PO::ArgumentParser::ArgumentDescriptor&, std::basic_string_view<char, std::char_traits<char> >)@Base
 WasmEdge::PO::ArgumentParser::SubCommandDescriptor::consume_long_option(std::basic_string_view<char, std::char_traits<char> >)@Base
 ```

As far as I can tell, this is a side effect of commit 0b285f7bc01cc1797997da1e537161d31d7449c1, which commented out `CXX_VISIBILITY_PRESET` in cmake/Helper.cmake.

I assume this change was not intentional?

(Also see issue #1678 and d4f604b6a05b48a1ad4d9b4fb65c5b7314860a1b for prior work in this area).

### Reproduction steps

1. Build with options '...'
2. Execute with flags '....'
3. Execute with inputs '....'
4. Get error

### Components

CLI, C SDK

### WasmEdge Version or Commit you used

0.14.0

### Operating system information

Debian unstable

### Hardware Architecture

amd64

**Comment by hydai** [comment]
This change is not intentional. I created #3656 to fix it. Because we don't have a CI check for this, could you give some ideas on what we can do to avoid this error being made again?

**Comment by paravoid** [comment]
Thank you so much for the blazingly fast response! I don't have any good ideas on how to catch that kind of thing with CI :/ I'll think about it.

For what it's worth, the way I identified this is through `dpkg-gensymbols`, which happens as part of the Debian package build process.

**Comment by hydai** [comment]
And I found that the ffmpeg plugin will fail if we make `CXX_VISIBILITY_PRESET` hidden. I think this is the reason that the previous contributor for the ffmpeg plugin tried to disable it. I am digging into this plugin to check if there is something wrong because the current design of the plugin architecture should never use these symbols from our codebase.

**Comment by hydai** [comment]
For now, the first priority is to revert this regression. Since I don't have much time for the wasmedge-ffmpeg plugin, I will temporarily remove and disable all related CI workflows.

If anyone is interested in fixing the wasmedge-ffmpeg, please check #3658 for more details.

**Comment by paravoid** [comment]
> Thank you so much for the blazingly fast response! I don't have any good ideas on how to catch that kind of thing with CI :/ I'll think about it.

A belt-and-suspenders approach could perhaps be to replace the `WasmEdge::*` wildcard in `lib/api/libwasmedge.lds` with an explicit list of the (C++) symbols that you are supposed to export. This will mask, rather than identify underlying issues like this one, and it also won't catch C API symbols that are inadvertently exposed, but perhaps better than the alternative?

**Comment by hydai** [comment]
Let me keep this open until we have a good solution for detecting this problem.

**Comment by hydai** [comment]
To clarify the issue, I have created a standalone ticket to track the solution and prevent this issue from occurring again.
Let's move forward to #3743 instead of this since this issue is fixed in 0.14.1.
